framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/util/SsoDeployer.java

```diff
@@ -0,0 +1,94 @@
+        project.processTemplateAndCreateResources(OpenShiftTemplate.SSO_SECRET.getTemplateUrl(), Collections.emptyMap());
+        logger.info("Creating SSO image streams from " + OpenShiftConstants.getSSOImageStreams());
+        project.createResources(OpenShiftConstants.getSSOImageStreams());
+
+        logger.info("Processing template and createing resources from " + OpenShiftTemplate.SSO.getTemplateUrl().toString());
+        ssoEnvVariables.put(OpenShiftTemplateConstants.IMAGE_STREAM_NAMESPACE, project.getName());
+
+        project.processTemplateAndCreateResources(OpenShiftTemplate.SSO.getTemplateUrl(), ssoEnvVariables);
+
+        logger.info("Waiting for SSO deployment to become ready.");
+        ssoDeployment.waitForScale();
+
+        createRolesAndUsers(ssoDeployment.getUrl().toString() + "/auth", SSO_REALM, envVariables);
+
+        return ssoDeployment;
+    }
+
+    private static SsoDeployment createSsoDeployment(Project project) {
+        SsoDeploymentImpl ssoDeploymnet = new SsoDeploymentImpl(project);
+        ssoDeploymnet.setUsername(DeploymentConstants.getSsoAdminUser());
+        ssoDeploymnet.setPassword(DeploymentConstants.getSsoAdminPassword());
+
+        return ssoDeploymnet;
+    }
+
+    // RHPAM-1228
+    public static String createSsoEnvVariable(String url) {
+        String[] urlParts = url.split(":");
+        return urlParts[0] + ":" + urlParts[1] + "/auth";
+    }
+
+    private static void createRolesAndUsers(String authUrl, String realm, Map<String, String> envVariables) {
+        logger.info("Creating roles and users in SSO at URL {} in Realm {}", authUrl, realm);
+        SsoApi ssoApi = SsoApiFactory.getRestApi(authUrl, realm);
+
+        ssoApi.createRole(ADMIN);
+        ssoApi.createRole(KIE_SERVER);
+        ssoApi.createRole(REST_ALL);
+        ssoApi.createUser(envVariables.getOrDefault(OpenShiftTemplateConstants.KIE_ADMIN_USER, DeploymentConstants.getWorkbenchUser()),
+                envVariables.getOrDefault(OpenShiftTemplateConstants.KIE_ADMIN_PWD, DeploymentConstants.getWorkbenchPassword()),
+                Arrays.asList(ADMIN, KIE_SERVER, REST_ALL));
+        ssoApi.createUser(envVariables.getOrDefault(OpenShiftTemplateConstants.KIE_SERVER_CONTROLLER_USER, DeploymentConstants.getControllerUser()),
+                envVariables.getOrDefault(OpenShiftTemplateConstants.KIE_SERVER_CONTROLLER_PWD, DeploymentConstants.getControllerPassword()),
+                Arrays.asList(KIE_SERVER, REST_ALL));
+        ssoApi.createUser(envVariables.getOrDefault(OpenShiftTemplateConstants.KIE_SERVER_USER, DeploymentConstants.getKieServerUser()),
+                envVariables.getOrDefault(OpenShiftTemplateConstants.KIE_SERVER_PWD, DeploymentConstants.getKieServerPassword()),
+                Arrays.asList(KIE_SERVER));
+        ssoApi.createUser(envVariables.getOrDefault(OpenShiftTemplateConstants.BUSINESS_CENTRAL_MAVEN_USERNAME, MavenConstants.getMavenRepoUser()),
```

Comment:
What is the point of setting this user/password? AFAIK it belongs to the Nexus user which isn't covered by SSO.